src/BaseSocket.php

```diff
@@ -0,0 +1,106 @@
+<?php
+namespace KS;
+
+class BaseSocket
+{
+    protected $socket=-1;
+
+    public static function newRawSocket($socketDomain, $socketType, $socketProtocol)

```

Comment:
After looking at `InetSocket` and `UnixSocket`, my preference here would be EITHER:

1. Eliminate `newRawSocket`, rename `newSocket` to `create`, and make it abstract, forcing descendants to call `\socket_create` themselves; OR
2. Make `newRawSocket` protected, rename `newSocket` to `create` and make it abstract, forcing descendants to call `newRawSocket` themselves.

I don't see much advantage to abstracting the call to `\socket_create` here, since it's hard to imagine a situation in which creating a generic socket would ever be more complex than just calling the function itself.

In either of the two options above, the advantage would be that there would be _two_ fewer methods to use, document and maintain, and use would be much clearer, because your only options for creating a socket would be to call `UnixSocket::create` or `InetSocket::create`. In fact, we could probably even make the constructor protected so that sockets may _only_ be created through these public static methods.